hypershift-operator/controllers/nodepool/manifests.go

```diff
@@ -150,3 +151,24 @@ func TokenSecret(namespace, name, payloadInputHash string) *corev1.Secret {
 		},
 	}
 }
+
```

Comment:
let's move this into its own agent.go file in this package so the size of this file does not get out of hand as we introduce more platforms.